### PR TITLE
[IMP+FIX][14.0] shopinvader_import_image

### PR DESCRIPTION
**Pull request by ivantodorovich**
Description in commits

After:

image

### CODE REVIEW

File: shopinvader_import_image/models/import_image.py [hunk 1]
```diff
@@ -159,24 +166,21 @@ def _get_lines(self):
         lines = []
         with closing(io.BytesIO(self._read_csv())) as file_csv:
             reader = csv_reader(file_csv, delimiter=self.csv_delimiter)
-            headers = next(reader, None)
-
-            if headers != self.csv_header.split(self.csv_delimiter):
+            headers = self.csv_header.split(self.csv_delimiter)
```
Comment (simahawk) [inline shopinvader_import_image/models/import_image.py]: to be changed as per our chat to provide real mapping of fields :wink:

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by ivantodorovich** [comment]
Ready. Changed with a real mapping and dictreader

File: shopinvader_import_image/models/import_image.py [hunk 2]
```diff
@@ -157,27 +174,19 @@ def _read_csv(self):
...
-                    continue
-                default_code, tag_name, file_path = row
-                lines.append(
-                    {
-                        "default_code": default_code,
-                        "tag_name": tag_name,
-                        "file_path": file_path,
-                    }
-                )
+                line = {key: row.get(column) for key, column in mapping.items()}
```
Comment (simahawk) [inline shopinvader_import_image/models/import_image.py]: do not use get and let it raise a KeyError, catch that exception and raise a UserError "CSV schema imcompatible" or something similar

File: shopinvader_import_image/views/import_product_image_view.xml [hunk 3]
```diff
@@ -14,61 +28,65 @@
...
+                            <field
+                                name="external_csv_path"
+                                attrs="{
+                                        'required': [('source_type', '=', 'external_storage')],
+                                        'invisible': [('source_type', '!=', 'external_storage')],
+                                    }"
+                            />
+                            <field name="csv_delimiter" />
+                        </group>
+                        <group name="output" string="Import">
```
Comment (simahawk) [inline shopinvader_import_image/views/import_product_image_view.xml]:
```suggestion
                        <group name="destination" string="Destination">
```

**Review by simahawk — COMMENTED** [review]
can you post a screenshot?

**Comment by ivantodorovich** [comment]
Suggestions applied and posted a screenshot in the PR description :)

File: shopinvader_import_image/models/import_image.py [hunk 4]
```diff
@@ -157,27 +174,23 @@ def _read_csv(self):
...
-                        "tag_name": tag_name,
-                        "file_path": file_path,
-                    }
-                )
+                try:
+                    line = {key: row[column] for key, column in mapping.items()}
+                except KeyError as e:
+                    _logger.error(e)
+                    raise exceptions.UserError(_("CSV Schema Incompatible"))
+                if line:
```
Comment (simahawk) [inline shopinvader_import_image/models/import_image.py]: this IF seems useless

**Review by simahawk — APPROVED** [review]
waaaay better tnx!

File: shopinvader_import_image/models/import_image.py (hunk 4 above)
Comment (ivantodorovich) [reply shopinvader_import_image/models/import_image.py]: indeed!

**Review by sebalix — APPROVED** [review]

**Pull request merged**